### DISCUSSION TITLE
Installation Feels Convoluted?

### DISCUSSION THREAD

**Discussion by bdrelling** [kean/Pulse#45] [Q&A] [answered]
Hello, I'm trying to figure out how to get started using the macOS app. On your website, you mentioned that I can grab it from the latest Releases, but all that I can download seems to be source code. I know you recently rolled out Pulse Pro, so I thought maybe you removed the apps from the releases and moved it into Pulse Pro. I tried to navigate to `kean/PulsePro` like several links on your blog recommend, but they all 404. Then I figured it was because I wasn't a sponsor, so I chipped in a dollar to see if that would make a difference. Still 404s. Nothing actually says _how_ one gets access, or if the link is even valid.

I'm willing to pay more for Pulse Pro, but all of your instructions seem to be leading me around in circles. I'm so confused and I just want to be able to use your beautiful products.

Please improve your onboarding instructions.

**Comment by kean** [comment, marked as answer]
Hey, @bdrelling. Thanks for your feedback and for adding yourself as a sponsor!

I'm planning to improve the installation process and the documentation around it in the future. I spent all the time on the app itself and haven't updated all the related resources yet. In the future, all the info will be available in a clear way at https://kean.blog/pulse/home.

For now, I added the following note to the introductory post: "the repo is invite-based; invites are sent manually." Since it's a manual process, as you can imagine, there can be delays. I'm thinking about automating it.

Fortunately, the Pulse [framework](https://github.com/kean/Pulse) is completely open-source and is available for everyone. And the instructions on how to integrate it and use it are already available on the [official website](https://kean.blog/pulse/guides/overview). But Pulse Pro is going to continue being invite-only for now.

If something isn't clear, please feel free to open an issue or a discussion.

**Comment by bdrelling** [comment]
@kean Honestly, the one sentence about "the repo is invite-based; invites are sent manually" would have done wonders if I had that when I was going through the process, so that should be more than enough to help people figure out the Pulse Pro onboarding plan. 👍🏻 

I received the invite shortly afterwards, and in the time between trying to find it and receiving it, I was mostly in the dark. Everything before that point felt very confusing.

The other big issue with getting started: You mentioned that the Pulse framework has instructions on the official website, and that's great, but to be clear I'm only referring to the macOS App that you offer. On the [Pulse macOS](https://kean.blog/pulse/guides/pulse-macos) section of your Pulse guide, it states:

> I’m still sorting out the distribution. As of now, you can find a build in the latest GitHub release.

If I go to the [Pulse Releases](https://github.com/kean/Pulse/releases) page on GitHub, I can see that the previous two releases don't have the binary as part of the release.

If this is due to the fact that Pulse Pro is intended to replace the built binary, then please update your documentation accordingly. This also sent me on a bit of a wild goose chase wondering what repo I was missing or what download link I couldn't find. I almost ended up building from source rather than simply sponsoring you--which I'm happy to do for all the effort you've put into the apps.

Thank you for your hard work! I'm just starting to use the framework now and, if it continues this well, I absolutely plan to raise my sponsorship amount. 👍🏻

**Reply by kean** [reply]
Yeah, you are absolutely right. The [Pulse macOS](https://kean.blog/pulse/guides/pulse-macos) page also desperately needs to be updated. I haven't figured out whether I'm going to deprecate the previous macOS app or continue maintaining it in its current form. I'm thinking about deprecating because having not one, but two (almost completely different) macOS apps feels like a bit too much.

**Reply by bdrelling** [reply]
I think it's extremely reasonable to remove the macOS app and focus on Pulse Pro. 

My current understanding is that, if I really wanted to, I could create a barebones macOS App and throw PulseUI into it and get the freely available macOS App, right? Whereas Pulse Pro is a customized experienced intended for better capabilities?

Assuming I'm not too far off, I think focusing on Pulse Pro but improving the distribution would be a better priority than anything on the free macOS app. Is the macOS App Store not a good option? (Apologies, I've never actually had to deploy a macOS app before!)

Developers that want to use this technology can afford to send you money, and I think the app you've created has a beautiful UI/UX so far, so it's definitely worth the cost.

I like Proxyman's model: They didn't invent it or anything, but it's something like $30/yr and you get all updates within that first year. If you just stick to free Proxyman, it's very usable, but you lose some very good features for "power usage" like pinning and tabs and so on. TablePlus has a similar model.

I really enjoy those models because the moment I start using the app and feel how seamless the experience is, I immediately want the power user features. Additionally, as a standalone user for a small app, I can cut cost by not paying for Proxyman, but for a professional use case (like at my company), we buy a Proxyman license for all developers regardless as to whether or not they need the power user features. 

Hope this feedback helps! It's all just opinion, but maybe it'll add some perspective as you make a decision around distribution. Whatever you decide, I'm excited to see it! Good luck!

**Reply by kean** [reply]
Thanks for your input. App Store is ... challenging. It's a bit early, and I'm not sure I'm at the stage where I can commit to developing a commercial app. It's more of a hobby project that I use for learning purposes, and I'm happy to earn a bit via donations. Chances are Apple will sherlock it in Xcode 14.

**Comment by sgruby** [comment]
If you keep the framework as open source and keep Pulse Pro closed source, I'd suggest keeping the previous Mac app around so that people can build/modify the previous app and view logs.

**Reply by kean** [reply]
Thanks, Scott. I think you are right. It doesn't feel right to take away the existing functionality. Plus, one of the ideas behind the project was to showcase SwiftUI across all Apple platforms. Yes, the previous macOS app is simple, but it's still a native macOS app.

**Comment by bdrelling** [comment]
In that case, I think it’s absolutely fair to keep distribution behind the
sponsorship wall. As long as the instructions are cleared up, I can’t see
there being any issue!

**Reply by kean** [reply]
In the meantime, I updated all of the docs. Hope it's now more clear. Thanks again for your feedback!
